### ISSUE TITLE
Permission issue causes sqlite database creation to fail

### ISSUE THREAD

**Issue by numanair** [TheNickOfTime/handbrake-web#149]
Running the provided docker-compose.yml, even with `user: 1000:1000 ` causes the /data volume to be created with root ownership. This prevents the sqlite3 database from being created, preventing the server from starting. A simple `chown` allows it to start up.

**Comment by TheNickOfTime** [comment]
Hey did the directories that you mapped in the compose file exist prior to you starting the container, or were they created automatically on the container's startup?

**Comment by starstruct** [comment]
Same issue, this is the error I get

```
root@hb:~/docker# docker compose up handbrake-server
[+] Running 1/0
 ✔ Container handbrake-web-server  Created                                                                                                 0.0s 
Attaching to handbrake-web-server
handbrake-web-server  | 
handbrake-web-server  | > handbrake-web@0.0.0 prod
handbrake-web-server  | > tsx src/server.ts
handbrake-web-server  | 
handbrake-web-server  | 
handbrake-web-server  | /handbrake-web/server/node_modules/better-sqlite3/lib/database.js:69
handbrake-web-server  | 		[util.cppdb]: { value: new addon.Database(filename, filenameGiven, anonymous, readonly, fileMustExist, timeout, verbose || null, buffer || null) },
handbrake-web-server  | 		                       ^
handbrake-web-server  | SqliteError: unable to open database file
handbrake-web-server  |     at new Database (/handbrake-web/server/node_modules/better-sqlite3/lib/database.js:69:26)
handbrake-web-server  |     at Database (/handbrake-web/server/src/scripts/database/database.ts:10:37)
handbrake-web-server  |     at Object.<anonymous> (/handbrake-web/server/src/scripts/database/database.ts:14:17)
handbrake-web-server  |     at Module._compile (node:internal/modules/cjs/loader:1546:14)
handbrake-web-server  |     at Object.transformer (/handbrake-web/server/node_modules/tsx/dist/register-CCR7NebB.cjs:3:1101)
handbrake-web-server  |     at Module.load (node:internal/modules/cjs/loader:1317:32)
handbrake-web-server  |     at Module._load (node:internal/modules/cjs/loader:1127:12)
handbrake-web-server  |     at TracingChannel.traceSync (node:diagnostics_channel:315:14)
handbrake-web-server  |     at wrapModuleLoad (node:internal/modules/cjs/loader:217:24)
handbrake-web-server  |     at Module.require (node:internal/modules/cjs/loader:1339:12) {
handbrake-web-server  |   code: 'SQLITE_CANTOPEN'
handbrake-web-server  | }
handbrake-web-server  | 
handbrake-web-server  | Node.js v22.6.0
handbrake-web-server exited with code 1
```

**Comment by starstruct** [comment]
> Same issue, this is the error I get
> 
> ```
> root@hb:~/docker# docker compose up handbrake-server
> [+] Running 1/0
>  ✔ Container handbrake-web-server  Created                                                                                                 0.0s 
> Attaching to handbrake-web-server
> handbrake-web-server  | 
> handbrake-web-server  | > handbrake-web@0.0.0 prod
> handbrake-web-server  | > tsx src/server.ts
> handbrake-web-server  | 
> handbrake-web-server  | 
> handbrake-web-server  | /handbrake-web/server/node_modules/better-sqlite3/lib/database.js:69
> handbrake-web-server  | 		[util.cppdb]: { value: new addon.Database(filename, filenameGiven, anonymous, readonly, fileMustExist, timeout, verbose || null, buffer || null) },
> handbrake-web-server  | 		                       ^
> handbrake-web-server  | SqliteError: unable to open database file
> handbrake-web-server  |     at new Database (/handbrake-web/server/node_modules/better-sqlite3/lib/database.js:69:26)
> handbrake-web-server  |     at Database (/handbrake-web/server/src/scripts/database/database.ts:10:37)
> handbrake-web-server  |     at Object.<anonymous> (/handbrake-web/server/src/scripts/database/database.ts:14:17)
> handbrake-web-server  |     at Module._compile (node:internal/modules/cjs/loader:1546:14)
> handbrake-web-server  |     at Object.transformer (/handbrake-web/server/node_modules/tsx/dist/register-CCR7NebB.cjs:3:1101)
> handbrake-web-server  |     at Module.load (node:internal/modules/cjs/loader:1317:32)
> handbrake-web-server  |     at Module._load (node:internal/modules/cjs/loader:1127:12)
> handbrake-web-server  |     at TracingChannel.traceSync (node:diagnostics_channel:315:14)
> handbrake-web-server  |     at wrapModuleLoad (node:internal/modules/cjs/loader:217:24)
> handbrake-web-server  |     at Module.require (node:internal/modules/cjs/loader:1339:12) {
> handbrake-web-server  |   code: 'SQLITE_CANTOPEN'
> handbrake-web-server  | }
> handbrake-web-server  | 
> handbrake-web-server  | Node.js v22.6.0
> handbrake-web-server exited with code 1
> ```

This happens with and without creating the directories manually

**Comment by numanair** [comment]
> Hey did the directories that you mapped in the compose file exist prior to you starting the container, or were they created automatically on the container's startup?

Only the parent folder existed before hand. The container created the data folder.

**Comment by TheNickOfTime** [comment]
I'll have to look into this to see if there is anything I can do, but I know I've seen this kind of behavior when running other docker containers. I'm guessing if a mapped directory doesn't exist, docker creates it - probably with root permissions as well.

It may just be the reality of running a docker container rootless - but I'll put a task in for myself later to see if I can account for it or at the very least do a permissions check and throw a descriptive error.

**Comment by TheNickOfTime** [comment]
Development task for this is #151 - target is the 0.7 release :)

**Comment by TheNickOfTime** [comment]
This issue should be resolved in the v0.7.0 release, which is now available!
